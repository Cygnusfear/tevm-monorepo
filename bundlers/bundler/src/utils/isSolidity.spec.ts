import { isSolidity, parseSolidityFile } from '.'
import { describe, expect, it } from 'vitest'
import { runSync } from 'effect/Effect'

describe(isSolidity.name, () => {
	it('should return true for .sol files', () => {
		const files = [
			'foo.sol',
			'foo.t.sol',
			'foo.s.sol',
			'foo/bar.sol',
			'foo/bar/baz.sol',
			'.sol/bar/baz.sol',
		]
		files.forEach((file) => {
			expect(isSolidity(file)).toBe(true)
		})
	})

	it('should return false for non .sol files', () => {
		const files = [
			'sol/.sol',
			'foo/bar.ts',
			'bar.sol.ts',
			'sol',
			'.sol/.sol',
			'.sol',
		]
		files.forEach((file) => {
			expect(isSolidity(file)).toBe(false)
		})
	})
})

describe('parseSolidityFile', () => {
	it('should return a Effect<never, never, string> when parsing a valid solidity file', () => {
		const files = [
			'foo.sol',
			'foo.t.sol',
			'foo.s.sol',
			'foo/bar.sol',
			'foo/bar/baz.sol',
			'.sol/bar/baz.sol',
		]
		files.forEach((file) => {
			expect(runSync(parseSolidityFile(file))).toBe(file)
		})
	})
	it('should return a Effect<never, InvalidSolidityFileError, string> when parsing an invalid solidity file', () => {
		const files = [
			'sol/.sol',
			'foo/bar.ts',
			'bar.sol.ts',
			'sol',
			'.sol/.sol',
			'.sol',
		]
		files.forEach((file) => {
			expect(() => runSync(parseSolidityFile(file))).toThrowErrorMatchingSnapshot()
		})
	})
})
