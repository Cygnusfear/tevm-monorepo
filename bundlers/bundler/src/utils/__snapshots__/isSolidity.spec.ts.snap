// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseSolidityFile > should return a Effect<never, InvalidSolidityFileError, string> when parsing an invalid solidity file 1`] = `
"InvalidSolidityFileError: Provided value sol/.sol is not a valid Address
https://evmts.dev/reference/errors"
`;

exports[`parseSolidityFile > should return a Effect<never, InvalidSolidityFileError, string> when parsing an invalid solidity file 2`] = `
"InvalidSolidityFileError: Provided value foo/bar.ts is not a valid Address
https://evmts.dev/reference/errors"
`;

exports[`parseSolidityFile > should return a Effect<never, InvalidSolidityFileError, string> when parsing an invalid solidity file 3`] = `
"InvalidSolidityFileError: Provided value bar.sol.ts is not a valid Address
https://evmts.dev/reference/errors"
`;

exports[`parseSolidityFile > should return a Effect<never, InvalidSolidityFileError, string> when parsing an invalid solidity file 4`] = `
"InvalidSolidityFileError: Provided value sol is not a valid Address
https://evmts.dev/reference/errors"
`;

exports[`parseSolidityFile > should return a Effect<never, InvalidSolidityFileError, string> when parsing an invalid solidity file 5`] = `
"InvalidSolidityFileError: Provided value .sol/.sol is not a valid Address
https://evmts.dev/reference/errors"
`;

exports[`parseSolidityFile > should return a Effect<never, InvalidSolidityFileError, string> when parsing an invalid solidity file 6`] = `
"InvalidSolidityFileError: Provided value .sol is not a valid Address
https://evmts.dev/reference/errors"
`;
